WordPress/gutenberg
# Replace function expressions with named function or arrow expression

Related (facilitates): #22610

This pull request seeks to replace the majority of [function expressions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/function) throughout the codebase, with an equivalent (named) [function declaration](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function) or [arrow function expression](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions).

This serves a few purposes:

- Code consistency
- Reduce changes involved with the Prettier upgrade in #22610 
- Encourage named functions where appropriate, since it helps dramatically in error stack traces, debugger call stacks, and performance audits to quickly identify a function by name (vs. "(anonymous)").

**Implementation Notes:**

While the changes are numerous, I expected them to be straight-forward. It was complicated by the fact that arrow functions have a few distinct differences from function expressions, which are used intentionally in some cases (availability of [`arguments`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments) and difference in [`this` binding](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this)). I've made a best effort to verify against regressions to intentional references to `arguments` and `this` which could be affected by these changes, and have avoided a few areas of mixed usage altogether (e.g. `block-editor/src/utils/transform-styles/ast`).

**Testing Instructions:**

Ensure all tests and end-to-end tests pass:

```
npm test
```

```
npm run test-e2e
```

Perform spot testing to identify potential regressions. Particular attention should be paid to affected code.
